### PR TITLE
Update 05_ddp.md

### PR DESCRIPTION
**Pull request by XuHu0529**
sbp示例代码中，DistributedSampler封装使dataloader进行分布式数据划分

### CODE REVIEW

File: cn/docs/parallelism/05_ddp.md [hunk 1]
```diff
@@ -35,8 +35,10 @@
         download=True,
     )
 
+    sampler = flow.utils.data.distributed.DistributedSampler(training_data)
+
     train_dataloader = flow.utils.data.DataLoader(
-        training_data, BATCH_SIZE, shuffle=True
+        training_data, BATCH_SIZE, shuffle=(sampler is None), sampler=sampler
```
Comment (doombeaker) [inline cn/docs/parallelism/05_ddp.md:41]: @Liuzy0908 @CPFLAME @Ldpe2G 胡旭同学说，这样改过后，还是会有 acc 和单卡不对齐的问题。
可以帮忙分析下吗

> 在这里加上DistributedSampler之后，好像和单卡还是有差别的，loss差别看着不大，epoch5的时候acc要差2个点，是不是哪里不太对啊

Comment (doombeaker) [reply cn/docs/parallelism/05_ddp.md:41]:
```suggestion
        training_data, BATCH_SIZE, shuffle=False, sampler=sampler
```

先把这个 shuffle 设死为 `False`，排除随机的干扰，试试 acc 是不是会一致看看呢？

**Comment by doombeaker** [comment]
代码没问题后。记得英文版（`en` 目录下）的对应文章中的代码也要更新下。

File: cn/docs/parallelism/05_ddp.md [hunk 2]
```diff
@@ -35,8 +35,10 @@
         download=True,
     )
 
+    sampler = flow.utils.data.distributed.DistributedSampler(training_data)
```
Comment (Ldpe2G) [inline cn/docs/parallelism/05_ddp.md:38]:
```suggestion
    sampler = flow.utils.data.distributed.DistributedSampler(training_data, shuffle=True)
```

**Comment by doombeaker** [comment]
我发现原文档中已经提示过 sampler 的问题

https://docs.oneflow.org/master/parallelism/05_ddp.html#distributedsampler

image


所以最开始的例子还是保持不变吧，在 05_ddp.html#distributedsampler 那节做修改，添加一个有 distributedsampler，所以单机单卡和分布式训练效果会一致的例子。

**Comment by doombeaker** [comment]
> 我发现原文档中已经提示过 sampler 的问题
> 
> https://docs.oneflow.org/master/parallelism/05_ddp.html#distributedsampler
> 
> image
> 
> 所以最开始的例子还是保持不变吧，在 05_ddp.html#distributedsampler 那节做修改，添加一个有 distributedsampler，所以单机单卡和分布式训练效果会一致的例子。

这一条要求好像还没有更新

File: cn/docs/parallelism/05_ddp.md [hunk 3]
```diff
@@ -88,6 +91,8 @@
     y = y.to_global(placement=PLACEMENT, sbp=S0)
 ```
 
+- 需要注意的是，在进行分布式并行训练时，代码中规定的`BATCH_SIZE`为每一台机器的本地值而非`GLOBAL_BATCH_SIZE`,故上述代码单机双卡`BATCH_SIZE=64`的训练效果与单机单卡`BATCH_SIZE=128`一致。
```
Comment (doombeaker) [inline cn/docs/parallelism/05_ddp.md:94]:
```suggestion
- 需要注意的是，在进行分布式并行训练时，代码中规定的 `BATCH_SIZE` 为每一台机器的本地值而非`GLOBAL_BATCH_SIZE`,故上述代码单机双卡 `BATCH_SIZE=64` 的训练效果与单机单卡 `BATCH_SIZE=128` 一致。
```

中英文之间、中文和数字之间要有空格。

Comment (doombeaker) [reply cn/docs/parallelism/05_ddp.md:94]: 其实我觉得这句不用加这里，因为它如果懂 global tensor，应该自己懂这个道理。
如果真要解释，是不是把 global tensor 那篇文章多做解释，解释下各种 sbp 下，to global 后的 global tensor 的形状。

**Review by doombeaker — COMMENTED** [review]
现在好像还有些 comment 的意见没有处理，请查看下，不管修改不修改，都回复下。

另外，如果是 ready for review 的状态，请提供下在线预览或者编译效果截图。

File: cn/docs/parallelism/05_ddp.md (hunk 1 above)
Comment (brandonliu2) [reply cn/docs/parallelism/05_ddp.md:41]: > 先把这个 shuffle 设死为 `False`，排除随机的干扰，试试 acc 是不是会一致看看呢？

此设定对函数效果无影响

**Comment by brandonliu2** [comment]
> > 我发现原文档中已经提示过 sampler 的问题
> > https://docs.oneflow.org/master/parallelism/05_ddp.html#distributedsampler
> > image
> > 所以最开始的例子还是保持不变吧，在 05_ddp.html#distributedsampler 那节做修改，添加一个有 distributedsampler，所以单机单卡和分布式训练效果会一致的例子。
> 
> 这一条要求好像还没有更新

我的理解是，上面的那个sbp例子是有问题的，因为没有加distributedsampler，导致训练时两张卡拿到了一样的数据，global_tensor就没起效果，客户跑了之后就产生了疑问。

File: cn/docs/parallelism/05_ddp.md (hunk 3 above)
Comment (brandonliu2) [reply cn/docs/parallelism/05_ddp.md:94]: > 其实我觉得这句不用加这里，因为它如果懂 global tensor，应该自己懂这个道理。
> 如果真要解释，是不是把 global tensor 那篇文章多做解释，解释下各种 sbp 下，to global 后的 global tensor 的形状。

好的，global tensor的文档中已经有相应的tensor形状变化的解释以及例子。因为客户在微信聊天记录里问了一下这个`batch_size=64`是local还是global，我想着这里再解释一遍。

File: cn/docs/parallelism/05_ddp.md (hunk 2 above)
Comment (brandonliu2) [reply cn/docs/parallelism/05_ddp.md:38]: > Suggested change
>     sampler = flow.utils.data.distributed.DistributedSampler(training_data)
>     sampler = flow.utils.data.distributed.DistributedSampler(training_data, shuffle=True)

`shuffle=True`是参数的默认值，可以加上以示提醒。
